views.py

```diff
@@ -174,33 +127,10 @@ def user_detail(user_id):
 
     user = User.query.get_or_404(user_id)
     
-    num_sightings = 0
-    type_data = {}
-    
-    if user.sightings:
```

Comment:
So this edit only applies if `Sighting` table holds only unique values. If it doesn't, you might be able to use `.distinct()` don't quote me on the exact syntax, use documentation).

```
user.sightings.distinct(Pokemon.pokemon_id)
```